### ISSUE TITLE
Sporadic "Authorization is denied" error message in the event logs.

### ISSUE THREAD

**Issue by viloun** [OneIdentity/ActiveRoles-Safeguard-JIT-Access#4]
When requesting/checking access, it appears like this event message is thrown in the logs

```
An error occured while calling "HandleAccessRequestEvent"
OneIdentity.SafeguardDotNet.SafeguardDotNetException: Error returned from Safeguard API, Error: Unauthorized {"Code":60094,"Message":"Authorization is denied for this request.","InnerError":null}
   at OneIdentity.SafeguardDotNet.SafeguardConnection.InvokeMethodFull(Service service, Method method, String relativeUrl, String body, IDictionary`2 parameters, IDictionary`2 additionalHeaders)
   at OneIdentity.SafeguardDotNet.SafeguardConnection.InvokeMethod(Service service, Method method, String relativeUrl, String body, IDictionary`2 parameters, IDictionary`2 additionalHeaders)
   at OneIdentity.ARSGJitAccess.Common.SafeguardRestApiClient.GetAssetAccount(String assetAccountId)
   at OneIdentity.ARSGJitAccess.Service.Service.HandleAccessRequestEvent(String eventName, String eventBody)
   at OneIdentity.SafeguardDotNet.Event.EventHandlerRegistry.<>c__DisplayClass1_1.<HandleEvent>b__0()
```

Even though that error is thrown the Service still removes or grants access..

`Revoke access for: CN=SG_ TempDomAdmin1,OU=JIT,OU=Safeguard,DC=oneidentity,DC=demo. Set EDSVA-ARSGJitAccess = false.`

Event log XML: 

```
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="ARSGJitAccess" /> 
  <EventID Qualifiers="0">12321</EventID> 
  <Level>4</Level> 
  <Task>0</Task> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2021-04-07T19:37:24.987647400Z" /> 
  <EventRecordID>513835</EventRecordID> 
  <Channel>Application</Channel> 
  <Computer>adm02.oneidentity.demo</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>Revoke access for: CN=SG_ TempDomAdmin1,OU=JIT,OU=Safeguard,DC=oneidentity,DC=demo. Set EDSVA-ARSGJitAccess = false.</Data> 
  </EventData>
  </Event>
```

```
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="ARSGJitAccess" /> 
  <EventID Qualifiers="0">17616</EventID> 
  <Level>2</Level> 
  <Task>0</Task> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2021-04-07T19:37:24.237655900Z" /> 
  <EventRecordID>513834</EventRecordID> 
  <Channel>Application</Channel> 
  <Computer>adm02.oneidentity.demo</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>An error occured while calling "HandleAccessRequestEvent" OneIdentity.SafeguardDotNet.SafeguardDotNetException: Error returned from Safeguard API, Error: Unauthorized {"Code":60094,"Message":"Authorization is denied for this request.","InnerError":null} at OneIdentity.SafeguardDotNet.SafeguardConnection.InvokeMethodFull(Service service, Method method, String relativeUrl, String body, IDictionary`2 parameters, IDictionary`2 additionalHeaders) at OneIdentity.SafeguardDotNet.SafeguardConnection.InvokeMethod(Service service, Method method, String relativeUrl, String body, IDictionary`2 parameters, IDictionary`2 additionalHeaders) at OneIdentity.ARSGJitAccess.Common.SafeguardRestApiClient.GetAssetAccount(String assetAccountId) at OneIdentity.ARSGJitAccess.Service.Service.HandleAccessRequestEvent(String eventName, String eventBody) at OneIdentity.SafeguardDotNet.Event.EventHandlerRegistry.<>c__DisplayClass1_1.<HandleEvent>b__0()</Data> 
  </EventData>
  </Event>
```

**Comment by aamburn** [comment]
Hi Viloun,

I've been able to reproduce this by allowing the SPP Access Token to expire. We use the SafeguardDotNet PersistentEventListener, which refreshes itself when the access token expires, however the connection used for other API queries (like GetAssetAccount in the stacktrace you provided) is not refreshed. 

That's why we continued to receive the notification and call into ARS, even when we get an authorization error to SPP. 

I'm working on a fix.
